Mathlib/Geometry/Manifold/Elaborators.lean

```diff
@@ -0,0 +1,460 @@
+/-
+Copyright (c) 2025 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot, Michael Rothgang
+-/
+import Mathlib.Geometry.Manifold.ContMDiff.Defs
+import Mathlib.Geometry.Manifold.MFDeriv.Defs
+import Mathlib.Geometry.Manifold.Traces
+
+/-!
+# Elaborators for differential geometry
+
+This file defines custom elaborators for differential geometry, to allow for more compact notation.
+There are two classes of elaborators. The first provides more compact notation for differentiability
+and continuous differentiability on manifolds, including inference of the model with corners.
+All these elaborators are scoped to the `Manifold` namespace. They allow writing
+- `MDiff f` for `MDifferentiable I J f`
+- `MDiffAt f x` for `MDifferentiableAt I J f x`
+- `MDiff[u] f` for `MDifferentiableOn I J f u`
+- `MDiffAt[u] f x` for `MDifferentiableWithinAt I J f u x`
+- `CMDiff n f` for `ContMDiff I J n f`
+- `CMDiffAt n f x` for `ContMDiffAt I J n f x`
+- `CMDiff[u] n f` for `ContMDiffOn I J n f u`
+- `CMDiffAt[u] n f x` for `ContMDiffWithinAt I J n f u x`,
+- `mfderiv[u] f x` for `mfderivWithin I J f u x`,
+- `mfderiv% f x` for `mfderiv I J f x`.
+
+In each of these cases, the models with corners are inferred from the domain and codomain of `f`.
+The search for models with corners uses the local context and is (almost) only syntactic, hence
+hopefully fast enough to always run.
+
+This has no dedicated support for product manifolds (or product vector spaces) yet;
+adding this is left for future changes. (It would make need to make a choice between e.g. the
+trivial model with corners on a product `E × F` and the product of the trivial models on `E` and
+`F`).
```

Comment:
Indicate that in these settings the elaborators should be avoided?

Reply:
Good idea, done!